### PR TITLE
10745 add a specific country selection for dn reports on dev test

### PR DESCRIPTION
**Pull request by IlyaIzr**

### CODE REVIEW

File: src/features/reports/atoms/tableAtom.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 
 type TableState = {
   sortIndex: number;
   thead?: string[];
-  data?: string[][] | null | 'sorting';
+  data?: string[][] | null;
   ascending: boolean | null;
   initialData?: string[][];
+  isSorting?: boolean;
+  _sortDefault?: string[][];
```
Comment (propakov) [inline src/features/reports/atoms/tableAtom.ts]: why do you use this _variableName syntax here? 
it's usually used to show that we wont use this variable

**Comment by propakov** [comment]
Screenshot 2022-07-11 at 16 55 23
Idk why, but placeholder overlaps cursor in input

File: src/features/reports/atoms/tableAtom.ts (hunk 1 above)
Comment (propakov) [reply src/features/reports/atoms/tableAtom.ts]: as it's the table data, maybe smth like defaultSortedData would be more suitable here?

File: src/features/reports/atoms/tableAtom.ts [hunk 2]
```diff
@@ -94,9 +99,33 @@ export const tableAtom = createAtom(
...
+    onAction('search', ({ query, columnIndexes }) => {
+      if (!query || !columnIndexes.length || !state.initialData)
+        return (state = { ...state, data: state.initialData });
 
-      state = { ...state, data: sorted };
+      const filtered = state.initialData.filter((row) => {
+        // match all register
+        let matched = false;
+        columnIndexes.forEach((index) => {
+          row[index].toLocaleLowerCase().includes(query.toLocaleLowerCase()) &&
```
Comment (propakov) [inline src/features/reports/atoms/tableAtom.ts]: It looks like you continue looping even if you found what's needed.
I would suggest to do `for of` and `break` right after you do (matched = true).

File: src/features/reports/components/ReportInfo/ReportInfo.tsx [hunk 3]
```diff
@@ -84,6 +85,9 @@ export function ReportInfo() {
         </div>
       )}
       <div className={clsx(!reportResource.data && styles.invisible)}>
+        {report?.searchable_columns_indexes?.length && (
+          <Searchbar searchIndexes={report?.searchable_columns_indexes} />
```
Comment (propakov) [inline src/features/reports/components/ReportInfo/ReportInfo.tsx]:
```suggestion
          <Searchbar searchIndexes={report.searchable_columns_indexes} />
```

File: src/features/reports/atoms/tableAtom.ts (hunk 1 above)
Comment (IlyaIzr) [reply src/features/reports/atoms/tableAtom.ts]: there's no way in reatom to declare private variable like in a class syntax, so we're using underscore to mark that it should be private
I'll implement your naming suggestion though

**Review by propakov — APPROVED** [review]
🔥

**Pull request merged**